lib/Reconciler.lua

```diff
@@ -323,7 +335,7 @@ function Reconciler._reconcileInternal(instanceHandle, newElement)
 
 		-- Cancel the old ref before we make changes. Apply the new one after.
 		if refChanged and oldRef then
-			oldRef(nil)
+			applyRef(oldRef, nil)
```

Comment:
We need to update the branch on line 349 that applies `newRef` as well! It's calling `newRef` as a function.